### PR TITLE
Update parameter values for ST_Point constructor docs

### PR DESCRIPTION
**Pull request by sheajamba**
## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Documentation

> How would you describe this change to a non-technical end user or system administrator?

Clarifies order in which longitude and latitude double values get passed to ST_Point constructor. 

## Documentation

( ) No documentation is needed.
(x) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

(x) No release notes entries required.
( ) Release notes entries required with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/functions/geospatial.rst [hunk 1]
```diff
@@ -90,7 +90,7 @@ Constructors
     Array elements must not be ``NULL`` or empty.
     The returned geometry may not be simple and may contain duplicate points if input array has duplicates.
 
-.. function:: ST_Point(double, double) -> Point
+.. function:: ST_Point(longitude, latitude) -> Point
```
Comment (mosabua) [inline docs/src/main/sphinx/functions/geospatial.rst:93]: This has to stay as the data type. Maybe we have to clarify in the description.. see how this is done in other functions in this document or elsewhere in the trino docs

Comment (martint) [reply docs/src/main/sphinx/functions/geospatial.rst]: ST_point and other geometry related functions don’t operate on longitude and latitude. They work on points on an Euclidean plane. 

You need to use the SphericalGeography related functions to use longitude and latitude.

File: docs/src/main/sphinx/functions/geospatial.rst [hunk 2]
```diff
@@ -92,7 +92,8 @@ Constructors
 
 .. function:: ST_Point(double, double) -> Point
 
-    Returns a geometry type point object with the given coordinate values.
+    Returns a geometry type point object with the given coordinate values in the
+    form (longitude, latitude).
```
Comment (martint) [inline docs/src/main/sphinx/functions/geospatial.rst:96]: This is actually incorrect. This function returns a point on a plane. The choice of whether to use it to represent a longitude/latitude coordinate is up to the user and context dependent.

**Review by martint — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/functions/geospatial.rst (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/functions/geospatial.rst:96]: @sheajamba can update this. Is there some information we can add on SphericalGeography functions and what we support in that context .. doesnt seem like we have much in the current docs..

**Pull request closed without merging**